src/main/java/teammates/common/datatransfer/attributes/NotificationAttributes.java

```diff
@@ -0,0 +1,407 @@
+package teammates.common.datatransfer.attributes;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.SanitizationHelper;
+import teammates.storage.entity.Notification;
+
+/**
+ * The data transfer object for the notification.
+ */
+public class NotificationAttributes extends EntityAttributes<Notification> {
+
+    private String notificationId;
+    private Instant startTime;
+    private Instant endTime;
+    private NotificationStyle style;
+    private NotificationTargetUser targetUser;
+    private String title;
+    private String message;
+    private boolean shown;
+    private Instant createdAt;
+    private Instant updatedAt;
```

Comment:
`createdAt` and `updatedAt` can probably be transient?

Reply:
Since the admin page for editing notifications has the feature of sorting by `createdAt`, we think `createdAt` could not be transient. `updatedAt` has been made transient and other comments have been addressed in this PR #11738. Thank you!